lib/ansible/modules/network/fortimanager/fmgr_device.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community"
+}
+
+DOCUMENTATION = '''
+---
+module: fmgr_device
+version_added: "2.8"
+author: Andrew Welsh, Luke Weighall
+short_description: Add or remove device
+description:
+  - Add or remove a device or list of devices from FortiManager Device Manager using jsonrpc API
+
+options:
+  adom:
+    description:
+      - The ADOM the configuration should belong to.
+    required: true
+    default: root
+  host:
+    description:
+      - The FortiManager's Address.
+    required: true
+  username:
+    description:
+      - The username used to authenticate with the FortiManager.
+    required: false
+  password:
+    description:
+      - The password associated with the username account.
+    required: false
+  state:
+    description:
+      - The desired state of the specified object.
+      - absent will delete the object if it exists.
+      - present will create the configuration if needed.
+    required: false
+    default: present
+    choices: ["absent", "present"]
+
+  device_username:
+    description:
+      - The username of the device being added to FortiManager.
+    required: false
+  device_password:
+    description:
+      - The password of the device being added to FortiManager.
+    required: false
+  device_ip:
+    description:
+      - The IP of the device being added to FortiManager.
```

Comment:
Can this support both IPv4 and IPv6 addresses?

Reply:
Yes. Made a change to reflect this.